### PR TITLE
[2946][ADD] sale_commission_seller_agent

### PR DESCRIPTION
**Pull request by AungKoKoLin1997**
[2946](https://www.quartile.co/web#id=2946&action=771&model=project.task&view_type=form&menu_id=505)

### CODE REVIEW

File: sale_commission_seller_agent/models/sale_order_line.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+    @api.depends("product_id")
+    def _compute_agent_ids_by_product(self):
+        for record in self:
+            if not record.commission_free:
+                # propose the seller of the product in agent line
+                if record.product_id.seller_id and record.product_id.seller_id.agent:
+                    record.agent_ids = [
+                        (0, 0, record._prepare_agent_vals(record.product_id.seller_id))
+                    ]
```
Comment (yostashiro) [inline sale_commission_seller_agent/models/sale_order_line.py]:
```suggestion
            if record.commission_free:
                continue
            # propose the seller of the product in agent line
            if record.product_id.seller_id and record.product_id.seller_id.agent:
                record.agent_ids = [
                    (0, 0, record._prepare_agent_vals(record.product_id.seller_id))
                ]
```

Just to reduce the if ladder.  A minor preference.

**Review by yostashiro — APPROVED** [review]
@kanda999 Please review the function and install the module in the test database.

File: sale_commission_seller_agent/models/sale_order_line.py [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+class SaleOrderLine(models.Model):
+    _inherit = "sale.order.line"
+
+    agent_ids = fields.One2many(compute="_compute_agent_ids_by_product")
+
+    @api.depends("product_id")
+    def _compute_agent_ids_by_product(self):
+        for record in self:
+            if not record.commission_free:
```
Comment (yostashiro) [inline sale_commission_seller_agent/models/sale_order_line.py]:
```suggestion
            if record.commission_free:
```

**Review by yostashiro — CHANGES_REQUESTED** [review]
Now you broke the logic. :(

**Review by yostashiro — APPROVED** [review]

**Review by kanda999 — APPROVED** [review]
Function review. It worked as expected.

**Comment by kanda999** [comment]
We need requirement.txt for this PR, but followed by https://github.com/qrtl/rmm-custom/pull/36
The latest commit is for using the gitaggregator.

**Comment by yostashiro** [comment]
@AungKoKoLin1997 Please rebase your branch and force-push.

**Comment by AungKoKoLin1997** [comment]
> @AungKoKoLin1997 Please rebase your branch and force-push.

Done.

**Comment by yostashiro** [comment]
@kanda999 Please go ahead and merge this at your convenience.

**Pull request merged**